alarms/alarms.go

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Abstract Machines
+// SPDX-License-Identifier: Apache-2.0
+
+package alarms
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/absmach/supermq/pkg/authn"
+)
+
+const (
+	SeverityMax uint8 = 100
+)
+
+var ErrInvalidSeverity = errors.New("invalid severity. Must be between 0 and 100")
+
+type Metadata map[string]interface{}
+
+// Alarm represents an alarm instance.
+type Alarm struct {
+	ID         string    `json:"id"`
+	RuleID     string    `json:"rule_id"`
+	Message    string    `json:"message"`
```

Comment:
Instead of a message, we should have more complex value, such as measurement (a string representing a key), value (numeric or string), unit (string), and cause (also a string). For example measurement="temperature", value=105, unit="C", cause="too hot" (senML format almost).